tests/pyccel/scripts/highorder_functions.py

```diff
@@ -0,0 +1,90 @@
+ # pylint: disable=missing-function-docstring, missing-module-docstring/
+from pyccel.decorators import types
+
+#$ header function f1(int)
+def f1(a):
+    return a
+
+#$ header function f2(int)
+def f2(a):
+    return a * 2
+
+#$ header function f3(int)
+def f3(a):
+    return a * 5
+
+#$ header function f4(int, real)
+def f4(a, b):
+    return a + b
+
+#$ header function f5(real, real, real)
+def f5(a, b, c):
+    return a * b + c
+
+#$ header function f6(int, int)
+def f6(a, b):
+    return a * 5 + b
+
+#$ header function f7(real, real)
+def f7(a, b):
+    return a * 5 + b
+
+#$ header function f8()
+def f8():
+    return 0.5
+
+#$ header function f9(real, real) results(real, real)
+def f9(x, y):
+    return x * y, x / y
+
+@types()
+def f10():
+    y = 0
+    if y:
+        pass
+
+#$ header function high_int_1((int)(int), int)
+def high_int_1(function, a):
+    x = function(a)
+    return x
```

Comment:
`function` is a reserved keyword in Fortran. I am a bit surprised that the generated code actually compiles!

Reply:
I stand corrected. The Fortran standard defines many **keywords**, but it does **not** define any **reserved words**. Hence calling a function `function` is totally legal, but it's definitely not a good practice.